ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PartyManagement.scala

```diff
@@ -183,4 +183,68 @@ final class PartyManagement(session: LedgerSession) extends LedgerTestSuite(sess
+  ) {
+    case Participants(Participant(alpha), Participant(beta)) =>
+      val alphaPartyHintAndDisplayName =
+        "PMGetPartiesIsLocal_" + Random.alphanumeric.take(10).mkString
+      val betaPartyHintAndDisplayName =
+        "PMGetPartiesIsLocal_" + Random.alphanumeric.take(10).mkString
+      for {
+        alphaLocalParty <- alpha.allocateParty(
+          partyIdHint = Some(alphaPartyHintAndDisplayName),
+          displayName = Some(alphaPartyHintAndDisplayName),
+        )
+        betaLocalParty <- beta.allocateParty(
+          partyIdHint = Some(betaPartyHintAndDisplayName),
+          displayName = Some(betaPartyHintAndDisplayName),
+        )
+        // only wait for parties on ledgers configured to know about each other's parties
+        _ <- alpha.waitForParties(Set(alpha, beta), Set(alphaLocalParty, betaLocalParty))
+        _ <- beta.waitForParties(Set(alpha, beta), Set(alphaLocalParty, betaLocalParty))
+        alphaPartyOnAlphaParticipant <- alpha.getParties(Seq(alphaLocalParty))
+        alphaPartyOnBetaParticipant <- beta.getParties(Seq(alphaLocalParty))
+        betaPartyOnAlphaParticipant <- alpha.getParties(Seq(betaLocalParty))
+        betaPartyOnBetaParticipant <- beta.getParties(Seq(betaLocalParty))
+      } yield {
+        alphaPartyOnAlphaParticipant.headOption match {
+          case None =>
+            Assertions.fail("Expected to find alpha party on alpha participant")
+          case Some(alphaLocal) =>
+            assert(alphaLocal.displayName == alphaPartyHintAndDisplayName && alphaLocal.isLocal)
+        }
+
+        betaPartyOnBetaParticipant.headOption match {
+          case None =>
+            Assertions.fail("Expected to find beta party on beta participant")
+          case Some(betaLocal) =>
+            assert(betaLocal.displayName == betaPartyHintAndDisplayName && betaLocal.isLocal)
+        }
+
+        // On multi-participant ledgers remote parties should either not be known to each
+        // other or should be marked as non-local.
+        if (alpha.endpointId != beta.endpointId) {
+          alphaPartyOnBetaParticipant.headOption match {
+            // distributed ledgers may choose not to publish parties across participants (e.g. privacy-enabled P-KVUtils)
+            case None =>
+            // but if they do, remote parties have to be marked non-local
+            case Some(alphaRemote) =>
+              assert(
+                alphaRemote.displayName == alphaPartyHintAndDisplayName && !alphaRemote.isLocal)
+          }
+
+          betaPartyOnAlphaParticipant.headOption match {
+            // distributed ledgers may choose not to publish parties across participants (e.g. privacy-enabled P-KVUtils)
+            case None =>
+            // but if they do, remote parties have to be marked non-local
+            case Some(betaRemote) =>
+              assert(betaRemote.displayName == betaPartyHintAndDisplayName && !betaRemote.isLocal)
+          }
+        }
+      }
+  }
```

Comment:
This seems a bit verbose for what it aims to do, can you check if something along the lines of https://github.com/digital-asset/daml/pull/6413 could do it? Feel free to either merge it into this PR or adopt the bits that you find relevant.

Reply:
Thanks for the help simplifying the test - much better now.